Source/ASVideoNode.mm

```diff
@@ -650,33 +649,29 @@ - (void)setMuted:(BOOL)muted
 
 - (void)play
 {
-  __instanceLock__.lock();
+  ASLockScopeSelf();
 
   if (![self isStateChangeValid:ASVideoNodePlayerStatePlaying]) {
-    __instanceLock__.unlock();
     return;
   }
 
   if (_player == nil) {
-    __instanceLock__.unlock();
-      [self setNeedsPreload];
-    __instanceLock__.lock();
+    ASUnlockScope(self);
+    [self setNeedsPreload];
   }
 
   if (_playerNode == nil) {
     _playerNode = [self constructPlayerNode];
 
-    __instanceLock__.unlock();
-      [self addSubnode:_playerNode];
-    __instanceLock__.lock();
-      
+    ASUnlockScope(self);
+    [self addSubnode:_playerNode];
```

Comment:
I'm not sure if it is important to re-lock before the setNeedsLayout, but the locking scope did change here (may want to add a { } scope around the addSubnode)

Reply:
Good catch, either way it's outside the scope of this diff 👍